test/integration/multicluster/multicluster-traffic/mc_traffic_test.go

```diff
@@ -0,0 +1,369 @@
+package multiclustertraffic
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/pkg/healthcheck"
+	"github.com/linkerd/linkerd2/pkg/version"
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var (
+	TestHelper *testutil.TestHelper
+	targetCtx  string
+	contexts   map[string]string
+)
+
+var (
+	tcpConnRE = regexp.MustCompile(
+		`tcp_open_total\{direction="outbound",peer="dst",target_addr="[0-9\.]+:[0-9]+",target_ip="[0-9\.]+",target_port="[0-9]+",tls="true",server_id="default\.linkerd-multicluster-statefulset\.serviceaccount\.identity\.linkerd\.cluster\.local",dst_control_plane_ns="linkerd",dst_namespace="linkerd-multicluster-statefulset",dst_pod="nginx-statefulset-0",dst_serviceaccount="default",dst_statefulset="nginx-statefulset"\} [1-9]\d*`,
+	)
+	httpReqRE = regexp.MustCompile(
+		`request_total\{direction="outbound",target_addr="[0-9\.]+:[0-9]+",target_ip="[0-9\.]+",target_port="[0-9]+",tls="true",server_id="default\.linkerd-multicluster-statefulset\.serviceaccount\.identity\.linkerd\.cluster\.local",dst_control_plane_ns="linkerd",dst_namespace="linkerd-multicluster-statefulset",dst_pod="nginx-statefulset-0",dst_serviceaccount="default",dst_statefulset="nginx-statefulset"\} [1-9]\d*`,
+	)
+)
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	// Before starting, get source context
+	contexts = TestHelper.GetMulticlusterContexts()
+	targetCtx = contexts["tgt"]
+	// Then, re-build clientset with context of target cluster instead of kube
+	// context inferred from environment.
+	if err := TestHelper.SwitchContext(targetCtx); err != nil {
+		out := fmt.Sprintf("Error running test: failed to switch Kubernetes client to context [%s]: %s\n", targetCtx, err)
+		os.Stderr.Write([]byte(out))
+		os.Exit(1)
+	}
+	// Block until viz deploy is running successfully in target cluster.
+	TestHelper.WaitUntilDeployReady(testutil.LinkerdVizDeployReplicas)
+	os.Exit(m.Run())
+}
+
+// TestGateways tests the `linkerd multicluster gateways` command by installing
+// three emojivoto services in target cluster and asserting the output against
+// the source cluster.
+func TestGateways(t *testing.T) {
+	if err := TestHelper.SwitchContext(contexts["tgt"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["tgt"], err)
+	}
```

Comment:
Do we need to switch the context again here? This should run after `TestMain` above which already switched to `tgt` context.

Reply:
Probably not. I've added it in there to be sure we start off with the right context, I wasn't sure whether Go would start a go routine or something similar and mess up the state. I've removed it and we can re-add it if we experience flakiness.